app/services/index_deleter.rb

```diff
@@ -0,0 +1,37 @@
+class IndexDeleter
+  def initialize(args)
+    @index_alias = args[:index_alias]
+    @client = args[:client]
+    @logger = args[:logger]
+  end
+
+  def run
+    aliases = client.indices.get_aliases.keys
+    indexes_to_be_deleted = prepare_indexes_for_deletion aliases
+    delete indexes_to_be_deleted
+  rescue => e
+    msg = "Failed to delete alias.\n#{e.message}"
+    logger.error msg
+    Raven.capture_error msg
+  end
+
+  private
+
+  attr_reader :client, :index_alias, :logger
+
+  def prepare_indexes_for_deletion(aliases)
+    # Ensure that the three most recent indexes are kept
```

Comment:
this comment is probably a sign that you should rename your method to be more explicit about the intent.